File: samples/nativemessage/src/test/java/com/example/myapplication/SealedTest.java
package com.example.myapplication;

import org.junit.Test;

public class SealedTest {
    @Test
    public void testSealed(){
        Base b = SealedTestKt.getSealed2();
        b.getName();
//        Sealed2 s = (Sealed2)b;
//        System.out.println(s.getText());

        if (b instanceof Sealed1) {
            System.out.println("SEALED1");
        }else  if (b instanceof Sealed2) {
            System.out.println("SEALED2");
        }
    }
}

File: samples/nativemessage/src/main/java/com/example/myapplication/SealedTest.kt
package com.example.myapplication

sealed class Base(
    val name : String
)

class Sealed1(
    name : String,
    val id : Int
) : Base(name)
class Sealed2(
    name : String,
    val text : String
) : Base(name)

fun getSealed1() : Base = Sealed1("test", 1)
fun getSealed2() : Base = Sealed2("test", "")